### ISSUE TITLE
Custom rule IDs are flagged in .markdownlint.json when opened in editor

### ISSUE THREAD

**Issue by nschonni** [DavidAnson/vscode-markdownlint#127]
Came across this as part of https://github.com/microsoft/vscode-docs-authoring/pull/747
When you disable custom rules in your markdownlint config like
```json
{
    "default": true,
    "DOCSMD001": false,
    "DOCSMD010": false,
    "DOCSMD011": false,
    "MD001": false,
    "MD004": false,
...
```
The custom rule IDs are flagged as `Property DOCSMD001 is not allowed.`
Is there a proposed why to make the schema validation happy for custom rules?

**Comment by DavidAnson** [comment]
The schema is bundled with the extension and explicitly lists all the permitted values. I don't think there is a way to allow this, but I will check again before I close this as "unfixable".

**Comment by nschonni** [comment]
Yeah, I figured it might not be fixable, but I was wondering if there was a way to "contribute" to the schema with the values registerd through https://github.com/DavidAnson/vscode-markdownlint#markdownlintcustomrules.

**Comment by DavidAnson** [comment]
This is the relevant contribution point: https://code.visualstudio.com/api/references/contribution-points#contributes.jsonValidation

Today, it's a static file in the extension install directory. I can imagine a hack where the extension updates the file dynamically, but it seems like a bad idea. What might be possible is for the schema itself to allow for unspecified values. That may be expressible, but I need to review the schema syntax definition first.

**Comment by DavidAnson** [comment]
Looks like I CAN allow this by setting `additionalProperties` to `true` - or better, allowing `boolean` and `object` as with existing properties. I'll want to update the unit tests to be strict, but the public schema can be flexible.

https://json-schema.org/understanding-json-schema/reference/object.html

**Comment by nschonni** [comment]
OK, cool! If it doesn't work, that's fine too, since it's just a warning in the terminal, but I figured it would confuse people :)

**Comment by DavidAnson** [comment]
This will be fixed in the next release by this update to the library: https://github.com/DavidAnson/markdownlint/commit/6061cce169ecf8820cd8cd0b9f1c6570ebb3dafe

**Comment by DavidAnson** [comment]
Should be fixed in `0.37.0`.

**Comment by nschonni** [comment]
Confirmed it isn't giving the warning in the Problems tab. Thanks!
